stdlib/sql/bigquery.go

```diff
@@ -0,0 +1,179 @@
+package sql
+
+import (
+	"database/sql"
+	"math/big"
+	"time"
+
+	"cloud.google.com/go/civil"
+	_ "github.com/bonitoo-io/go-sql-bigquery"
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/codes"
+	"github.com/influxdata/flux/execute"
+	"github.com/influxdata/flux/internal/errors"
+	"github.com/influxdata/flux/values"
+)
+
+// Google BigQuery support.
+// Notes:
+// * data types
+//   https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types
+// * connection string
+//   https://github.com/bonitoo-io/go-sql-bigquery#connection-string
+
+type BigQueryRowReader struct {
+	Cursor      *sql.Rows
+	columns     []interface{}
+	columnTypes []flux.ColType
+	columnNames []string
+	sqlTypes    []*sql.ColumnType
+	NextFunc    func() bool
+	CloseFunc   func() error
+}
+
+// Next prepares BigQueryRowReader to return rows
+func (m *BigQueryRowReader) Next() bool {
+	if m.NextFunc != nil {
+		return m.NextFunc()
+	}
+	next := m.Cursor.Next()
+	if next {
+		columnNames, err := m.Cursor.Columns()
+		if err != nil {
+			return false
+		}
+		m.columns = make([]interface{}, len(columnNames))
+		columnPointers := make([]interface{}, len(columnNames))
+		for i := 0; i < len(columnNames); i++ {
+			columnPointers[i] = &m.columns[i]
+		}
+		if err := m.Cursor.Scan(columnPointers...); err != nil {
+			return false
+		}
+	}
+	return next
+}
+
+func (m *BigQueryRowReader) GetNextRow() ([]values.Value, error) {
+	row := make([]values.Value, len(m.columns))
+	for i, column := range m.columns {
+		switch value := column.(type) {
+		case bool, int64, float64, string:
+			row[i] = values.New(value)
+		case civil.Date:
+			row[i] = values.NewString(value.String())
+		case civil.DateTime:
```

Comment:
So far I have been mapping datetime types without TZ info to Flux strings, leaving formatting as is. You are right that values of such types are somewhat useless in Flux, but without timezone info, they cannot be RFC3339 formatted.  And I guess it be wrong to assume some timezone for datetime values we do not know what they represent, wouldn't it?

I can think of adding a proprietary connection string option (to all DBs supported by Flux that have such types without TZ) that would allow user to specify TZ to be used for these types, and when this would be set, we could map these types to either `time` or at least use RFC3399-formatted string.

Reply:
Ok, I like the idea of a special connection string option. If it becomes an issue perhaps we can do that. Otherwise, I think this is fine for now.